ddtrace/contrib/pytest/plugin.py

```diff
@@ -306,111 +310,132 @@ def _get_test_class_hierarchy(item):
-            test_class_hierarchy = _get_test_class_hierarchy(item)
-            if test_class_hierarchy:
-                span.set_tag_str(test.CLASS_HIERARCHY, test_class_hierarchy)
-            if hasattr(item, "dtest") and isinstance(item.dtest, DocTest):
-                span.set_tag_str(test.SUITE, "{}.py".format(item.dtest.globs["__name__"]))
-            else:
-                span.set_tag_str(test.SUITE, test_suite_span.get_tag(test.SUITE))
-
-        span.set_tag_str(test.TYPE, SpanTypes.TEST)
-        span.set_tag_str(test.FRAMEWORK_VERSION, pytest.__version__)
-
-        if item.location and item.location[0]:
-            _CIVisibility.set_codeowners_of(item.location[0], span=span)
-
-        # We preemptively set FAIL as a status, because if pytest_runtest_makereport is not called
-        # (where the actual test status is set), it means there was a pytest error
-        span.set_tag_str(test.STATUS, test.Status.FAIL.value)
+    is_skipped_by_itr = [
+        marker
+        for marker in item.iter_markers(name="skip")
+        if "reason" in marker.kwargs and marker.kwargs["reason"] == SKIPPED_BY_ITR
+    ]
 
-        # Parameterized test cases will have a `callspec` attribute attached to the pytest Item object.
-        # Pytest docs: https://docs.pytest.org/en/6.2.x/reference.html#pytest.Function
-        if getattr(item, "callspec", None):
-            parameters = {"arguments": {}, "metadata": {}}  # type: Dict[str, Dict[str, str]]
-            for param_name, param_val in item.callspec.params.items():
-                try:
-                    parameters["arguments"][param_name] = encode_test_parameter(param_val)
-                except Exception:
-                    parameters["arguments"][param_name] = "Could not encode"
-                    log.warning("Failed to encode %r", param_name, exc_info=True)
-            span.set_tag_str(test.PARAMETERS, json.dumps(parameters))
-
-        markers = [marker.kwargs for marker in item.iter_markers(name="dd_tags")]
-        for tags in markers:
-            span.set_tags(tags)
-        _store_span(item, span)
-
-        if coverage_enabled():
-            from ddtrace.internal.ci_visibility.coverage import cover
-
-            with cover(span, root=str(item.config.rootdir)):
-                yield
-        else:
+    if is_skipped_by_itr:
+        yield
+    else:
+        test_session_span = _extract_span(item.session)
+
+        pytest_module_item = _find_pytest_item(item, pytest.Module)
+        pytest_package_item = _find_pytest_item(pytest_module_item, pytest.Package)
+
+        test_module_span = _extract_span(pytest_package_item)
+        if pytest_package_item is not None and test_module_span is None:
+            if test_module_span is None:
+                test_module_span = _start_test_module_span(pytest_package_item)
+
+        test_suite_span = _extract_span(pytest_module_item)
+        if pytest_module_item is not None and test_suite_span is None:
```

Comment:
When would `pytest_module_item is None` not be some kind of error state? And should `_extract_span` above be done on `pytest_module_item` before checking that it isn't `None`?

Reply:
1. For runtime generated tests, I'm not sure a module would be required
2. `_extract_span` would just return `None` if it's passed `None`